client/filejsonstore/filejsonstore.go

```diff
@@ -0,0 +1,153 @@
+package client
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/theupdateframework/go-tuf/client"
+	"github.com/theupdateframework/go-tuf/internal/fsutil"
+	"github.com/theupdateframework/go-tuf/util"
+)
+
+// ErrNotJSON is returned when a metadata operation is attempted
+// against a file that does not seem to be a JSON file
+// (e.g. does not end in .json, case sensitive).
+var ErrNotJSON = errors.New("file is not in JSON format")
+
+// ErrToPermissive is returned when the metadata directory, or a metadata
+// file has too permissive file mode bits set
+var ErrTooPermissive = errors.New("permissions are too permissive")
+
+const (
+	// user:  rwx
+	// group: r-x
+	// other: ---
+	dirCreateMode = os.FileMode(0750)
+	// user:  rw-
+	// group: r--
+	// other: ---
+	fileCreateMode = os.FileMode(0640)
+)
+
+// FileJSONStore represents a local metadata cache relying on raw JSON files
+// as retrieved from the remote repository.
+type FileJSONStore struct {
+	mtx     sync.RWMutex
+	baseDir string
+}
+
+var _ client.LocalStore = (*FileJSONStore)(nil)
+
+// NewFileJSONStore returns a new metadata cache, implemented using raw JSON
+// files, stored in a directory provided by the client.
+// If the provided directory does not exist on disk, it will be created.
+// The provided metadata cache is safe for concurrent access.
+func NewFileJSONStore(baseDir string) (*FileJSONStore, error) {
+	var f = FileJSONStore{
```

Comment:
Can we initialize as a pointer? Now that there is a mutex field, the only valid way to use this type is via a pointer type, since otherwise the mutex is copied.

```suggestion
	f := &FileJSONStore{
```

Reply:
Yeah. The result are the same though as the pointer escapes this function, and all methods receives the target by pointer. But making it a pointer in the first place could be more clear.